grabbit/src/main/groovy/com/twcable/grabbit/ClientJobStatus.groovy

```diff
@@ -46,12 +49,22 @@ class ClientJobStatus {
         this.jcrNodesWritten = jcrNodesWritten
     }
 
-
-    public static ClientJobStatus get(@Nonnull JobExplorer explorer, @Nonnull Long jobExecutionId) {
+    /**
+     * Queries the status of a job
+     * @param explorer job explorer to query
+     * @param jobExecutionId the ID to search
+     * @throws {@link NonExistentJobException} if job can not be found. {@link IllegalArgumentException} if explorer or jobExecutionId are null
+     * @return {@link ClientJobStatus}
+     */
+    static ClientJobStatus get(@Nonnull JobExplorer explorer, @Nonnull Long jobExecutionId) {
```

Comment:
While `IllegalArgumentException` is such a standard precondition exception it can be assumed, `NonExistentJobException` is not: It really should be declared in the method signature.